tests/test_api.py

```diff
@@ -660,6 +572,124 @@ def test_create_get_delete_namespace_auth(testclient):
     assert r.status == schema.APIStatus.ERROR
 
 
+def _crud_common(testclient, auth, method, route, json=None, data_pred=None):
+    if auth:
+        testclient.login()
+
+    if json is not None:
+        response = method(route, json=json)
+    else:
+        response = method(route)
+
+    if auth:
+        response.raise_for_status()
+    else:
+        assert response.status_code == 403
+
+    r = schema.APIResponse.parse_obj(response.json())
+    if auth:
+        assert r.status == schema.APIStatus.OK
+        if data_pred is None:
+            assert r.data is None
+        else:
+            assert data_pred(r.data) is True
+    else:
+        assert r.status == schema.APIStatus.ERROR
+
+
+@pytest.mark.parametrize("auth", [True, False])
+def test_update_namespace_metadata(testclient, auth):
+    namespace = f"filesystem"
+    make_request = partial(_crud_common, testclient=testclient, auth=auth)
+
+    make_request(
+        method=testclient.put,
+        route=f"api/v1/namespace/{namespace}/metadata",
+        json={"test_key1": "test_value1", "test_key2": "test_value2"},
+    )
```

Comment:
Because I removed one parameter from the metadata handler (since I had to split it up), this doesn't require passing the "metadata" key in JSON as before.